lsp-mode.el

```diff
@@ -7833,36 +7826,42 @@ reported according to `flycheck-check-syntax-automatically'."
         (lsp--flycheck-level level tags)
       level)))
 
+(defvar-local lsp--diagnostics-modified? nil)
+
 (defun lsp--flycheck-start (checker callback)
   "Start an LSP syntax check with CHECKER.
 
 CALLBACK is the status callback passed by Flycheck."
-  (->> (lsp--get-buffer-diagnostics)
-       (-map (-lambda (diag)
-               (flycheck-error-new
-                :buffer (current-buffer)
-                :checker checker
-                :filename buffer-file-name
-                :line (1+ (lsp-diagnostic-line diag))
-                :column (1+ (lsp-diagnostic-column diag))
-                :message (lsp-diagnostic-message diag)
-                :level (lsp--flycheck-calculate-level diag)
-                :id (lsp-diagnostic-code diag)
-                :end-column (-> diag
+  (when lsp--diagnostics-modified?
```

Comment:
do we need to check that here? It seems to me that once we are at that point we should report unconditionally. What happens when I explicitly call M-x flycheck-buffer after modified?

Reply:
> do we need to check that here?

No, it will remove it. After submitting the PR, I thought about the case you mention and I do not think flycheck has a cache.